src/utils/typeguards.ts

```diff
@@ -1,10 +1,136 @@
-import { SharedModels } from "@kontent-ai/management-sdk";
+import { ContentTypeModels, ContentTypeSnippetModels, SharedModels, TaxonomyModels } from "@kontent-ai/management-sdk";
+import { z } from "zod";
 
 import { spotlightInUseErrorCode } from "../constants/responseCodes.js";
+import {
+  CountLimitation,
+  DefaultElementValue,
+  DependsOn,
+  ExternalIdReference,
+  MaximumTextLength,
+  ObjectReference,
+  ValidationRegex,
+} from "../modules/sync/types/patchOperation.js";
 
 export const notNull = <T>(arg: T | null): arg is T => arg !== null;
 
 export const notNullOrUndefined = <T>(arg: T | undefined | null): arg is T => arg !== undefined && arg !== null;
 
-export const isSpotlightInUseError = (err: any): err is SharedModels.ContentManagementBaseKontentError =>
-  err.errorCode === spotlightInUseErrorCode;
+export const isSpotlightInUseError = (err: unknown): err is SharedModels.ContentManagementBaseKontentError =>
+  z.object({
+    errorCode: z.literal(spotlightInUseErrorCode),
+  }).safeParse(err).success;
+
+const elementTypesEnum = z.enum([
+  "text",
+  "rich_text",
+  "number",
+  "multiple_choice",
+  "date_time",
+  "asset",
+  "modular_content",
+  "taxonomy",
+  "url_slug",
+  "guidelines",
+  "snippet",
+  "custom",
+  "subpages",
+]);
+
+const elementSchema: z.ZodType = z.object({
+  type: elementTypesEnum,
+});
+
+const taxonomySchema: z.ZodType<TaxonomyModels.IAddTaxonomyRequestModel> = z.object({
+  name: z.string(),
+  terms: z.lazy(() => taxonomySchema.array()),
+  codename: z.string().optional(),
+  external_id: z.string().optional(),
+});
+
+const baseTypeSchema = z.object({
+  name: z.string(),
+  elements: elementSchema.array(),
+  external_id: z.string().optional(),
+  codename: z.string().optional(),
+});
+
+const snippetSchema: z.ZodType<ContentTypeSnippetModels.IAddContentTypeSnippetData> = baseTypeSchema;
+
+const contentTypeSchema: z.ZodType<ContentTypeModels.IAddContentTypeData> = baseTypeSchema.extend({
+  content_groups: z.object({
+    name: z.string(),
+    codename: z.string().optional(),
+    external_id: z.string().optional(),
+  }).array(),
+});
+
+const objectReferenceSchema: z.ZodType<ObjectReference> = z.object({
+  codename: z.string(),
+  id: z.string().optional(),
+  external_id: z.string().optional(),
+});
+
+const externalIdReferenceSchema: z.ZodType<ExternalIdReference> = z.object({
+  external_id: z.string(),
+});
+
+export const isTaxonomyData = (obj: unknown): obj is TaxonomyModels.IAddTaxonomyRequestModel =>
```

Comment:
isAddTaxonomyData. There is also another type TaxonomyModels.Taxonomy, so the name of the function can be confusing. Please check also the others

Reply:
all typeguards renamed accordingly